platform-api/che-core-api-infrastructure-local/src/test/java/org/eclipse/che/api/local/PredefinedRecipeLoaderTest.java

```diff
@@ -0,0 +1,64 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2015 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.api.local;
+
+import org.eclipse.che.api.core.ServerException;
+import org.eclipse.che.api.local.storage.PredefinedRecipeLoader;
+import org.eclipse.che.api.machine.server.dao.RecipeDao;
+import org.eclipse.che.api.machine.server.recipe.RecipeImpl;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.testng.MockitoTestNGListener;
+import org.testng.Assert;
+import org.testng.annotations.Listeners;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.mockito.Matchers.any;
+
+/**
+ * @author Anton Korneta
+ */
+@Listeners(MockitoTestNGListener.class)
+public class PredefinedRecipeLoaderTest {
+
+    private PredefinedRecipeLoader recipeLoader;
+
+    @Mock
+    private RecipeDao recipeDao;
+
+    @Test
+    public void shouldLoadPredictableRecipesFromValidJson() throws Exception {
+        recipeLoader = new PredefinedRecipeLoader("recipes.json", recipeDao);
+        List<RecipeImpl> result = new ArrayList<>();
+        Mockito.doAnswer(r -> result.add((RecipeImpl)r.getArguments()[0])).when(recipeDao).create(any());
+
+        recipeLoader.start();
+
+        Assert.assertEquals(result.size(), 2);
```

Comment:
I think in this case is better to use <code>verify</code> + <code>times()</code>